scipy/interpolate/src/__fitpack.cc

```diff
@@ -171,10 +171,97 @@ data_matrix( /* inputs */
     *nc = len_t - k - 1;
 }
 
+// Ref: https://github.com/scipy/scipy/blob/10f63b25d1e040cca3d7319dc2edff0c31ef8b7a/scipy/interpolate/fitpack/fpperi.f#L181-L195
+// and
+// Ref: https://github.com/scipy/scipy/blob/10f63b25d1e040cca3d7319dc2edff0c31ef8b7a/scipy/interpolate/fitpack/fpperi.f#L221-L238
+// For given set of knots, computes H1 and H2 for all the data points except the last.
+// Check the above link for semantics of H1 and H2.
```

Comment:
As mentioned, "check the above link" is not helpful. Please add comments in the style of the `data_matrix`.
Here ISTM, the computations is nearly the same as that in `data_matrix`, with additional something for H1 and H2 for each row. Please say it and explain the difference.